src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs

```diff
@@ -25,8 +26,18 @@ public static int Main(string[] args)
                 CompilerServerLogger.LogException(ex, "Error loading application settings");
             }
 
-            var controller = new DesktopBuildServerController(appSettings);
-            return controller.Run(args);
+            try
+            {
+                var controller = new DesktopBuildServerController(appSettings);
+                return controller.Run(args);
+            }
+            catch (TypeInitializationException ex) when (ex.InnerException is FileNotFoundException)
```

Comment:
It's possible that `csc` could throw a `TypeInitializationException`, although I haven't found such a case. In `VBCSCompiler.exe`, a `TypeInitializationException` results from a `FileNotFoundException` initializing `DesktopCompilerServerHost`.